ui/store/actions.js

```diff
@@ -1103,6 +1113,32 @@ export function updateMetamaskState(newState) {
       type: actionConstants.UPDATE_METAMASK_STATE,
       value: newState,
     });
+
+    // check that tx status has not changed and remove it from failed transactions if it has
+    const foundTx = currentNetworkTxList && currentNetworkTxList.find((currentTx, index) => {
```

Comment:
The `currentNetworkTxList` should always be set. If it's empty, it'll be an empty array.

```suggestion
    const foundTx = currentNetworkTxList.find((currentTx, index) => {
```

Reply:
Also... this only finds the first transaction in `currentNetworkTxList` that should be removed from `failedTransactionsToDisplay`. If multiple transactions were to be removed in one update, it'd miss the second one.

I think it would be better to loop through `failedTransactionsToDisplay` and check each one against `currentNetworkTxList` for state changes. That way we ensure we remove _all_ that should be removed. 

e.g. `const transactionIdsToRemove = failedTransactionsToDisplay.filter(...)`

Reply:
this was done in https://github.com/MetaMask/metamask-extension/commit/ac896b3f871fe530ea46745486dba0d78fbb2cec#diff-3c58b916cefff81e4927a9afd392acadbf4515f99f41783964a9851b7774791b